DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/EncodedBlocks.h

```diff
@@ -155,6 +155,48 @@ struct Block {
     return alignSize((_ndict + _ndata) * sizeof(W));
   }
 
+  void storeDict(int _ndict, const W* _dict)
+  {
+    if (nDict || nStored) {
+      throw std::runtime_error("trying to write in occupied block");
+    }
+    size_t sz = estimateSize(_ndict, 0);
+    assert(registry); // this method is valid only for flat version, which has a registry
+    assert(sz <= registry->getFreeSize());
+    assert((_ndict > 0) == (_dict != nullptr));
+    nDict = _ndict;
+    if (nDict) {
+      auto ptr = payload = reinterpret_cast<W*>(registry->getFreeBlockStart());
+      memcpy(ptr, _dict, _ndict * sizeof(W));
+      ptr += _ndict;
+    }
+  };
+
+  void storeData(int _ndata, const W* _data)
+  {
+    if (nStored) {
+      throw std::runtime_error("trying to write in occupied block");
+    }
+
+    size_t sz = estimateSize(nDict, _ndata);
```

Comment:
here should be ``size_t sz = estimateSize(0, _ndata);`` as we don't need to account for already stored dict size.